BuildSourceImage.sh

```diff
@@ -1,80 +1,968 @@
-    buildah config --created-by "/bin/sh -c #(nop) ADD file:$(cd ${CONTEXT_DIR}; tar cf - . | sha256sum -| cut -f1 -d' ') in /CONTEXT" ${SRC_CTR}
-    export IMG=$(buildah commit --omit-timestamp --rm ${SRC_CTR})
-    export SRC_CTR=$(buildah from ${IMG})
-fi
+unpack_img_umoci() {
+    local image_dir="${1}"
+    local unpack_dir="${2}"
+
+    _debug "unpackging with umoci"
+    # always assume we're not root I reckon
+    umoci unpack --rootless --image "${image_dir}" "${unpack_dir}" >&2
+    ret=$?
+    return $ret
+}
 
 #
-# If the caller specified a extra directory,
-# add it to the CONTEXT DIR in SRC IMAGE
+# copy an image from one location to another
 #
-if [ ! -z "${EXTRA_SRC_DIR}" ]; then
-    buildah add ${SRC_CTR} ${EXTRA_SRC_DIR} /EXTRA
-    buildah config --created-by "/bin/sh -c #(nop) ADD file:$(cd ${EXTRA_SRC_DIR}; tar cf - . | sha256sum -| cut -f1 -d' ') in /CONTEXT" ${SRC_CTR}
-    export IMG=$(buildah commit --omit-timestamp --rm ${EXTRA_SRC_CTR})
-    export SRC_CTR=$(buildah from ${IMG})
-fi
+push_img() {
+    local src="${1}"
+    local dst="${2}"
 
-# Cleanup and remove source container
-buildah rm ${SRC_CTR}
+    ## TODO: check for authfile, creds, and whether it's an insecure registry
+    skopeo copy --dest-tls-verify=false "$(ref_prefix ${src})" "$(ref_prefix ${dst})" # XXX for demo only
+    #skopeo copy "$(ref_prefix ${src})" "$(ref_prefix ${dst})"
+    ret=$?
+    return $ret
+}
 
 #
-# Add the final name to our image
+# sets up a basic new OCI layout, for an image with the provided (or default 'latest') tag
 #
-buildah tag $IMG $SRC_IMAGE
+layout_new() {
+    local out_dir="${1}"
+    local image_tag="${2:-latest}"
+
+    _mkdir_p "${out_dir}/blobs/sha256"
+    echo '{"imageLayoutVersion":"1.0.0"}' > "${out_dir}/oci-layout"
+    local config='
+{
+  "created": "'$(_date_ns)'",
+  "architecture": "amd64",
+  "os": "linux",
+  "config": {},
+  "rootfs": {
+    "type": "layers",
+    "diff_ids": []
+  }
+}
+    '
+    local config_sum=$(echo "${config}" | jq -c | tr -d '\n' | sha256sum | awk '{ ORS=""; print $1 }')
+    echo "${config}" | jq -c | tr -d '\n' > "${out_dir}/blobs/sha256/${config_sum}"
+
+    local mnfst='
+{
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "application/vnd.oci.image.config.v1+json",
+    "digest": "sha256:'"${config_sum}"'",
+    "size": '"$(_size ${out_dir}/blobs/sha256/${config_sum})"'
+  },
+  "layers": []
+}
+    '
+    local mnfst_sum=$(echo "${mnfst}" | jq -c | tr -d '\n' | sha256sum | awk '{ ORS=""; print $1 }')
+    echo "${mnfst}" | jq -c | tr -d '\n' > "${out_dir}/blobs/sha256/${mnfst_sum}"
+
+    echo '
+{
+  "schemaVersion": 2,
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:'"${mnfst_sum}"'",
+      "size": '"$(_size ${out_dir}/blobs/sha256/${mnfst_sum})"',
+      "annotations": {
+        "org.opencontainers.image.ref.name": "'"${image_tag}"'"
+      }
+    }
```

Comment:
```json
"platform": {
    "os": "linux"
}
```
?

Reply:
not required https://github.com/opencontainers/image-spec/blob/master/image-layout.md#indexjson-file